src/herder/HerderSCPDriver.cpp

```diff
@@ -711,22 +723,82 @@ HerderSCPDriver::combineCandidates(uint64_t slotIndex,
                 {
                     highestTxSet = cTxSet;
                     highest = sv.txSetHash;
+                    if (candidateCloseTimes.size() == 0)
+                    {
+                        throw std::logic_error(
+                            "Missing candidate closeTimes list");
+                    }
+                    highestTxSetIndex =
+                        make_optional<size_t>(candidateCloseTimes.size() - 1);
+                    if (candidateCloseTimes[*highestTxSetIndex] != sv.closeTime)
+                    {
+                        throw std::logic_error("Incorrect highestTxSet index "
+                                               "into candidate closeTimes");
+                    }
+                    if (svSigned)
+                    {
+                        if (sv.ext.v() != STELLAR_VALUE_SIGNED)
+                        {
+                            throw std::logic_error(
+                                "Unsigned candidate StellarValue");
+                        }
+                        highestTxSetSignature =
+                            make_optional<LedgerCloseValueSignature>(
+                                sv.ext.lcValueSignature());
+                    }
                 }
             }
         }
+
+        if (!highestTxSet)
```

Comment:
no need for this check: this would be a bug internal to that method

Reply:
Removed.